src/pkg/services/m365/m365_test.go

```diff
@@ -359,3 +362,240 @@ func (suite *m365UnitSuite) TestGetAllSites() {
+func TestDiscoveryIntgSuite(t *testing.T) {
+	suite.Run(t, &DiscoveryIntgSuite{
+		Suite: tester.NewIntegrationSuite(
+			t,
+			[][]string{tester.M365AcctCredEnvs}),
+	})
+}
+
+func (suite *DiscoveryIntgSuite) SetupSuite() {
+	suite.acct = tester.NewM365Account(suite.T())
+}
+
+func (suite *DiscoveryIntgSuite) TestUsers() {
+	t := suite.T()
+
+	ctx, flush := tester.NewContext(t)
+	defer flush()
+
+	errs := fault.New(true)
+
+	users, err := Users(ctx, suite.acct, errs)
+	assert.NoError(t, err, clues.ToCore(err))
+
+	ferrs := errs.Errors()
+	assert.Nil(t, ferrs.Failure)
+	assert.Empty(t, ferrs.Recovered)
+	assert.NotEmpty(t, users)
+}
+
+func (suite *DiscoveryIntgSuite) TestUsers_InvalidCredentials() {
+	table := []struct {
+		name string
+		acct func(t *testing.T) account.Account
+	}{
+		{
+			name: "Invalid Credentials",
+			acct: func(t *testing.T) account.Account {
+				a, err := account.NewAccount(
+					account.ProviderM365,
+					account.M365Config{
+						M365: credentials.M365{
+							AzureClientID:     "Test",
+							AzureClientSecret: "without",
+						},
+						AzureTenantID: "data",
+					},
+				)
+				require.NoError(t, err, clues.ToCore(err))
+
+				return a
+			},
+		},
+	}
+
+	for _, test := range table {
+		suite.Run(test.name, func() {
+			t := suite.T()
+
+			ctx, flush := tester.NewContext(t)
+			defer flush()
+
+			users, err := Users(ctx, test.acct(t), fault.New(true))
+			assert.Empty(t, users, "returned some users")
+			assert.NotNil(t, err)
+		})
+	}
+}
+
+func (suite *DiscoveryIntgSuite) TestSites() {
+	t := suite.T()
+
+	ctx, flush := tester.NewContext(t)
+	defer flush()
+
+	errs := fault.New(true)
+
+	sites, err := Sites(ctx, suite.acct, errs)
+	assert.NoError(t, err, clues.ToCore(err))
+
+	ferrs := errs.Errors()
+	assert.Nil(t, ferrs.Failure)
+	assert.Empty(t, ferrs.Recovered)
+	assert.NotEmpty(t, sites)
+}
+
+func (suite *DiscoveryIntgSuite) TestSites_InvalidCredentials() {
+	table := []struct {
+		name string
+		acct func(t *testing.T) account.Account
+	}{
+		{
+			name: "Invalid Credentials",
+			acct: func(t *testing.T) account.Account {
+				a, err := account.NewAccount(
+					account.ProviderM365,
+					account.M365Config{
+						M365: credentials.M365{
+							AzureClientID:     "Test",
+							AzureClientSecret: "without",
+						},
+						AzureTenantID: "data",
+					},
+				)
+				require.NoError(t, err, clues.ToCore(err))
+
+				return a
+			},
+		},
+		{
+			name: "Empty Credentials",
+			acct: func(t *testing.T) account.Account {
+				// intentionally swallowing the error here
+				a, _ := account.NewAccount(account.ProviderM365)
+				return a
+			},
+		},
+	}
+
+	for _, test := range table {
+		suite.Run(test.name, func() {
+			t := suite.T()
+
+			ctx, flush := tester.NewContext(t)
+			defer flush()
+
+			sites, err := Sites(ctx, test.acct(t), fault.New(true))
+			assert.Empty(t, sites, "returned some sites")
+			assert.NotNil(t, err)
+		})
+	}
+}
+
+func (suite *DiscoveryIntgSuite) TestUserInfo() {
```

Comment:
this is also a near duplicate of another test in this file though in this case the other one (can't comment on directly) should be removed since this has a superset of functionality